src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -24,49 +44,85 @@ class F extends ValidationTest {
     });
   }
 
-  createRenderPipeline(): GPURenderPipeline {
+  createRenderPipelineWithLayout(
+    device: GPUDevice,
+    bindGroups: Array<Array<GPUBindGroupLayoutEntry>>
+  ): GPURenderPipeline {
+    const shader = `
+      [[stage(vertex)]] fn vs_main() -> [[builtin(position)]] vec4<f32> {
+        return vec4<f32>(1.0, 1.0, 0.0, 1.0);
+      }
+
+      [[stage(fragment)]] fn fs_main() -> [[location(0)]] vec4<f32> {
+        return vec4<f32>(0.0, 1.0, 0.0, 1.0);
+      }
+    `;
+    const bindGroupLayouts = getBindGroupLayouts(this.device, bindGroups);
+
     const pipeline = this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts }),
       vertex: {
         module: this.device.createShaderModule({
-          code: `
-            [[block]] struct VertexUniforms {
-              transform : mat2x2<f32> ;
-            };
-            [[group(0), binding(0)]] var<uniform> uniforms : VertexUniforms;
-
-            [[stage(vertex)]] fn main(
-              [[builtin(vertex_index)]] VertexIndex : u32
-              ) -> [[builtin(position)]] vec4<f32> {
-              var pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
-                vec2<f32>(-1.0, -1.0),
-                vec2<f32>( 1.0, -1.0),
-                vec2<f32>(-1.0,  1.0)
-              );
-              return vec4<f32>(uniforms.transform * pos[VertexIndex], 0.0, 1.0);
-            }`,
+          code: shader,
         }),
-        entryPoint: 'main',
+        entryPoint: 'vs_main',
       },
       fragment: {
         module: this.device.createShaderModule({
-          code: `
-            [[block]] struct FragmentUniforms {
-              color : vec4<f32>;
-            };
-            [[group(1), binding(0)]] var<uniform> uniforms : FragmentUniforms;
-
-            [[stage(fragment)]] fn main() -> [[location(0)]] vec4<f32> {
-              return uniforms.color;
-            }`,
+          code: shader,
         }),
-        entryPoint: 'main',
+        entryPoint: 'fs_main',
         targets: [{ format: 'rgba8unorm' }],
       },
       primitive: { topology: 'triangle-list' },
     });
     return pipeline;
   }
 
+  createComputePipeline(
+    device: GPUDevice,
+    bindGroups: Array<Array<GPUBindGroupLayoutEntry>>
+  ): GPUComputePipeline {
+    const shader = `
+      [[stage(compute), workgroup_size(1, 1, 1)]]
+        fn main([[builtin(global_invocation_id)]] GlobalInvocationID : vec3<u32>) {
+      }
+    `;
+    const bindGroupLayouts = getBindGroupLayouts(this.device, bindGroups);
+    const pipeline = this.device.createComputePipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts }),
+      compute: {
+        module: this.device.createShaderModule({
+          code: shader,
+        }),
+        entryPoint: 'main',
+      },
+    });
+    return pipeline;
+  }
+
+  createRenderPipeline(device: GPUDevice): GPURenderPipeline {
```

Comment:
I think this helper is only used in `it_is_invalid_to_draw_in_a_render_pass_with_missing_bind_groups`?
But these BGL definitions are redundant - they're specified again below (at [2]) - so you could just use `createRenderPipelineWithLayout` directly there instead.